include/zephyr/drivers/satellite.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2022 Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SATELLITE_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SATELLITE_H_
+
+/**
+ * @file
+ * @brief Public Satellite APIs
+ * @defgroup satellite_api Satellite APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Satellite configuration.
+ */
+struct satellite_modem_config {
+	/** Frequency in Hz to use for transceiving */
+	uint32_t frequency;
+
+	/** TX-power in mW to use for transmission */
+	int16_t tx_power;
+};
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * For internal driver use only, skip these in public documentation.
+ */
+
+/**
+ * @typedef satellite_api_send_result_cb()
+ * @brief Callback API for to be return with TX status
+ */
+typedef void (*satellite_api_send_result_cb)(bool status);
+
+/**
+ * @typedef satellite_api_config()
+ * @brief Callback API for configuring the satellite module
+ *
+ * @see satellite_config() for argument descriptions.
+ */
+typedef int (*satellite_api_config)(const struct device *dev,
+			       struct satellite_modem_config *config);
+
+/**
+ * @typedef satellite_api_send_sync()
+ * @brief Callback API for sending data over satellite
+ *
+ * @see satellite_send_sync() for argument descriptions.
+ */
+typedef int (*satellite_api_send_sync)(const struct device *dev,
+			     uint8_t *data, uint32_t data_len);
+
+/**
+ * @typedef satellite_api_send_pool_async()
+ * @brief Callback API for sending pool of data asynchronously over satellite
+ * with delay between each transmission
+ *
+ * @see satellite_send_pool_async() for argument descriptions.
+ */
+typedef int (*satellite_api_send_pool_async)(const struct device *dev,
+				   uint8_t *data, uint32_t data_len,
+				   uint8_t number_of_send,
+				   k_timeout_t time_between_send,
+				   satellite_api_send_result_cb *result_cb);
+
+
+struct satellite_driver_api {
+	satellite_api_config config;
+	satellite_api_send_sync send_sync;
+	satellite_api_send_pool_async send_pool_async;
+};
```

Comment:
Hey trying to anticipate a bit the architecture WG discussion: I think the question would be if this API makes sense for a generic class of satellite communication device. Don't think we have existing expertise in the community about this so it'd be cool to know: is this something that potentially map to other type of satellite modems?

I see you already split the device specific APIs out in `drivers/satellite/kim1.h` (good!) so it seems like that has been thought out already.